reference_test.go

```diff
@@ -352,7 +359,11 @@ func TestWithNameFailure(t *testing.T) {
 			err:   ErrReferenceInvalidFormat,
 		},
 		{
-			input: strings.Repeat("a/", 128) + "a:tag",
```

Comment:
Wondering if there's still constraints (or if there should be) on maximum length of individual path components ("namespace", "repository-name"). At a quick glance, HTTP itself doesn't seem to define constraints; https://www.rfc-editor.org/rfc/rfc3986#section-3.3. Maybe it's fine to have the "total" alone. (it might be fun though to have a "valid" repository, but not able to use it because there's no space left for the _total_ length 🤔

Reply:
> it might be fun though to have a "valid" repository, but not able to use it because there's no space left for the total length

Can you please provide an example for this? I didn't completely understand it 😞